src/bandersnatch_storage_plugins/filesystem.py

```diff
@@ -262,3 +263,22 @@ def get_hash(self, path: PATH_TYPES, function: str = "sha256") -> str:
         digest = h.hexdigest()
         logger.debug(f"Calculated digest: {digest!s}")
         return str(h.hexdigest())
+
+    def get_file_size(self, path: PATH_TYPES) -> int:
+        if not isinstance(path, pathlib.Path):
+            path = pathlib.Path(path)
+        return path.stat().st_size
+
+    def get_upload_time(self, path: PATH_TYPES) -> datetime.datetime:
+        if not isinstance(path, pathlib.Path):
+            path = pathlib.Path(path)
+        return datetime.datetime.fromtimestamp(
+            path.stat().st_mtime, datetime.timezone.utc
+        )
+
+    def set_upload_time(self, path: PATH_TYPES, time: datetime.datetime) -> None:
+        """Set the upload time of a given **path**"""
+        if not isinstance(path, pathlib.Path):
+            path = pathlib.Path(path)
+        ts = time.timestamp()
+        os.utime(path, (ts, ts))
```

Comment:
Could the file ever not exist? Should we check for that or do we handle it catching the exceptions in the callers?

Reply:
At the moment, the only caller of these functions is `download_file()` in _mirror.py_, the `get_*()` ones are only called within an `if path.exist()` block, and `set_upload_time()`'s usage is extended to a situation right after a download is successful. If the file is gone afterwards then it signals something wrong is happening.